tsdb/head_append.go

```diff
@@ -458,6 +462,33 @@ func (a *headAppender) AppendHistogram(ref storage.SeriesRef, lset labels.Labels
 	return storage.SeriesRef(s.ref), nil
 }
 
+func validateHistogram(h *histogram.Histogram) error {
+	checkSpans := func(sign string, spans []histogram.Span, buckets []int64) error {
+		var spanBuckets uint32
```

Comment:
To keep it simple for comparison later (sum of uint32 can overflow unit32 althought it will not happen in practice here)
```suggestion
		var spanBuckets int
```